src/internal/request.ts

```diff
@@ -0,0 +1,34 @@
+import type * as http from 'node:http'
+import type * as https from 'node:https'
+import type * as stream from 'node:stream'
+import { pipeline } from 'node:stream'
+
+export async function request(
+  transport: typeof http | typeof https,
+  opt: https.RequestOptions,
+  body: Buffer | string | stream.Readable | null = null,
+): Promise<http.IncomingMessage> {
+  return new Promise<http.IncomingMessage>((resolve, reject) => {
+    const requestObj = transport.request(opt, (resp) => {
+      resolve(resp)
+    })
+
+    requestObj.on('error', (e: unknown) => {
+      reject(e)
+    })
+
+    if (body) {
+      if (!Buffer.isBuffer(body) && typeof body !== 'string') {
+        pipeline(body, requestObj, (err) => {
+          if (err) {
+            reject(err)
+          }
+        })
+      } else {
+        requestObj.write(body)
+      }
+    }
+
+    requestObj.end()
```

Comment:
I'm not very confident about this code.

Should this method only be called if `body` is not a stream?

Will it end request stream before `pipeline` could pump all content?

Reply:
We can fall back to the original implementation. we can focus on any refactoring or bug fixes later. this could help us in avoiding involved effort in review and test repeatedly . going back and forth would require repeated manual tests. this is my suggestion.